libyul/AsmAnalysis.cpp

```diff
@@ -324,7 +331,8 @@ std::vector<YulString> AsmAnalyzer::operator()(FunctionCall const& _funCall)
 			);
 		else if (
 			m_evmVersion.supportsTransientStorage() &&
-			_funCall.functionName.name == "tstore"_yulstring
+			_funCall.functionName.name == "tstore"_yulstring &&
+			ranges::find_if(m_errorReporter.errors(), isTransientStorageWarning) == ranges::end(m_errorReporter.errors())
```

Comment:
I'd make make this a generic helper in `ErrorReporter`.
I.e. `m_error_reporter.hasError(2394_error)` helper.